libs/algorithms/include/hpx/parallel/algorithms/transform.hpp

```diff
@@ -139,9 +324,9 @@ namespace hpx { namespace parallel { inline namespace v1 {
 
             template <typename ExPolicy, typename InIter, typename OutIter,
                 typename F, typename Proj>
-            HPX_HOST_DEVICE static std::pair<InIter, OutIter> sequential(
-                ExPolicy&& policy, InIter first, InIter last, OutIter dest,
-                F&& f, Proj&& proj)
+            HPX_HOST_DEVICE static util::in_out_result<InIter, OutIter>
+            sequential(ExPolicy&& policy, InIter first, InIter last,
```

Comment:
I think the implementation of transform needs to be changed such that it supports iterator/sentinel pairs. Could you add a test for this, please?